src/types.ts

```diff
@@ -89,8 +90,8 @@ type TabInfo = {
   id: number
   windowId: number
   active: boolean
-  url?: string
-  domain?: string
+  parsedUrl: null | ParseUrlSuccess
+  website: 'not fetched' | 'fetching' | 'failure' | Website
```

Comment:
I think we should have a separation between the status and the `Website` object, but this is for a different PR

Reply:
yeah, this is kind of gross, but we can live with it temporarily